### PR TITLE
`notify` event extension - fix tedge mapper child device create events

### PR DESCRIPTION
**Pull request by initard**
## Proposed changes

While trying to fix the tedge mapper to send child device created events, I thought it would be simpler to do this if `notify` was extended to allow directories to be separated from files. This PR proposes an extension of the `notify` crate to allow for DirectoryCreated, DirectoryDeleted, FileCreated, FileDeleted events, while using this to send 101 events for child device creation. 

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
#1520

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/core/tedge_mapper/src/c8y/dynamic_discovery.rs [hunk 1]
```diff
@@ -42,16 +42,17 @@ pub fn process_inotify_events(
...
                 event_type: EventType::Remove,
                 operation_name: operation_name.to_string(),
             })),
-            FileEvent::Created | FileEvent::Modified => Ok(Some(DiscoverOp {
+            FsEvent::FileCreated | FsEvent::Modified => Ok(Some(DiscoverOp {
                 ops_dir: parent_dir.to_path_buf(),
                 event_type: EventType::Add,
                 operation_name: operation_name.to_string(),
             })),
+            _ => Ok(None),  // FIXME: Should not get here I think?
```
Comment (didier-wenzek) [inline crates/core/tedge_mapper/src/c8y/dynamic_discovery.rs:55]: Why do you think that this case should not happen?

This looks good to me.

Comment (didier-wenzek) [reply crates/core/tedge_mapper/src/c8y/dynamic_discovery.rs]: Outdated

**Review by didier-wenzek — APPROVED** [review]
Approved to be merged on main.

Indeed, receiving explicit messages for directory events simplifies the code of the mapper.

**Comment by initard** [comment]
Fixes #1526

**Pull request merged**